test-utils/src/bin/falcon_runner.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Mechanisms to launch and control propolis VMs via falcon
+
+use anyhow::anyhow;
+use camino_tempfile::tempdir;
+use libfalcon::{unit::gb, Runner};
+use omicron_test_utils::dev::test_setup_log;
+use slog::info;
+use std::env;
+
+#[cfg(target_os = "illumos")]
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let args: Vec<_> = env::args().collect();
+    let logctx = test_setup_log("falcon_test_runner");
+    let log = logctx.log;
+    info!(log, "args = {:?}", args);
+    let name = "launchpad_mcduck";
+
+    let runner_name = format!("{}_runner", name);
+    let node_name = format!("{}_test_vm", name);
+    let mut d = Runner::new(&runner_name);
+    let vm = d.node(&node_name, "helios-2.0", 2, gb(8));
+
+    let falcon_dir = tempdir()?;
+    info!(log, "Setting falcon directory to {}", falcon_dir.path());
+    d.falcon_dir = falcon_dir.path().into();
+
+    let cargo_bay = tempdir()?;
+    info!(log, "Set cargo-bay to {} on host machine", cargo_bay.path());
+    let source_test_path = &args[1];
+    let test_name = &args[3];
+    let test_file_name = camino::Utf8Path::new(source_test_path)
+        .file_name()
+        .expect("Failed to get test file name");
+    let test_path = cargo_bay.path().to_path_buf().join(test_file_name);
+
+    info!(log, "Copying {source_test_path} to {test_path}");
+    std::fs::copy(camino::Utf8Path::new(source_test_path), &test_path)?;
+
+    d.mount(cargo_bay.path(), "/opt/cargo-bay", vm)?;
+
+    info!(log, "Launching test vm {node_name}");
+    d.launch().await?;
+    info!(log, "Launched test vm {node_name}");
+
+    info!(log, "Running test: {test_file_name}::{test_name}");
+    let run_test = format!(
+        "cd /opt/cargo-bay && chmod +x {test_file_name}; \
+        res=$?; \
+        if [[ $res -eq 0 ]]; then \
+          ./{test_file_name} --color never --exact {test_name} --nocapture; \
+          res=$?; \
+        fi; \
+        echo $res"
+    );
+    let out = d.exec(vm, &run_test).await?;
+
+    // The last line of our output contains the exit code
+    let exit_code_index = out.rfind('\n').unwrap();
+    let exit_code: u8 = (&out[exit_code_index + 1..]).parse().unwrap_or(255);
```

Comment:
Couple nitpicks here:

- If the `\n` character didn't exist in the output we might want to get better feedback from the test (via an `.expect` call, perhaps?)
- The indexing in `out` seems like it could easily go out-of-bounds -- what if `\n` was the last character emitted?